src/Microsoft.Framework.DesignTimeHost/ApplicationContext.cs

```diff
@@ -490,6 +497,34 @@ private bool PerformCompilation()
             return true;
         }
 
+        private bool PerformPluginWork()
+        {
+            if (_pluginRegistration.WasAssigned ||
+                _pluginWorkNeeded.WasAssigned ||
+                _filesChanged.WasAssigned ||
+                _rebuild.WasAssigned ||
+                _restoreComplete.WasAssigned ||
+                _sourceTextChanged.WasAssigned)
```

Comment:
Do you need to actually do something for all of these types of messages? That doesn't seem right. You only need to do work if there is a plugin registered.